### PR TITLE
Challenge 11: GET request using query string to populate filtered database

### PR DESCRIPTION
**Pull request by messat**
1. Added a query string to the database to filter by query
2. Added extensive error handling when the query is omitted or mismatched
3. Called next block err in the controller to handle the error instead of the error middleware handle block

### CODE REVIEW

**Comment by messat** [comment]
Merging branch challenge 11 to main branch

File: __tests__/newsApi.test.js [hunk 1]
```diff
@@ -409,4 +409,46 @@ describe('CORE Task 10: GET request /api/users', () => {
...
+    })
+  });
+  test('Status 404: GET request with a query that does not match the field property i.e. missspelt', () => {
+    return request(app)
+    .get('/api/articles?topicss=mitch')
+    .expect(404)
+    .then((res)=>{
+     expect(res.body.msg).toBe('404 Not Found')
+    })
+  });
```
Comment (MRombi) [inline __tests__/newsApi.test.js:453]: The topic paper is valid but has no articles so we can write another test with that in mind 🙂

File: __tests__/newsApi.test.js [hunk 2]
```diff
@@ -409,4 +409,46 @@ describe('CORE Task 10: GET request /api/users', () => {
...
+    return request(app)
+    .get('/api/articles?topic=dogs')
+    .expect(200)
+    .then((res)=>{
+      const articlesArr = res.body.articles
+     expect(articlesArr).toHaveLength(0)
+     expect(articlesArr).toEqual([])
+    })
+  });
+  test('Status 404: GET request with a query that does not match the field property i.e. missspelt', () => {
```
Comment (MRombi) [inline __tests__/newsApi.test.js:446]: We want a 404 when the query is potentially valid but does not exist in the database. For example /api/articles?topics=apple'

We either want to send a 400 or ignore when topicss is the query. In that sense this test could also be deleted 🙂

**Review by MRombi — COMMENTED** [review]
Good job on this initial attempt, please check the feedback when you can 🙂

**Pull request merged**